x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/condition/ScriptCondition.java

```diff
@@ -58,10 +58,6 @@ public Result execute(WatchExecutionContext ctx) {
     }
 
     public Result doExecute(WatchExecutionContext ctx) {
-        Map<String, Object> parameters = Variables.createCtxParamsMap(ctx, ctx.payload());
-        if (script.getParams() != null && !script.getParams().isEmpty()) {
-            parameters.putAll(script.getParams());
-        }
         WatcherConditionScript conditionScript = scriptFactory.newInstance(script.getParams(), ctx);
```

Comment:
you know what, i think thats right. It looks like `AbstractCompareCondition` does the same thing, passing in the `createCtxParamsMap()`. Whats interesting is that when this is actually executed, the `WatcherConditionScript` also has all of the values of the context in it. Ive found in a few spots where these contexts fail if you try to access any params that dont exist, due to it being null in a painless script. I think there might be some more code exploration I need to do before I feel safe with this. One other data point is that the `ExecutableScriptTransform`, the other place that painless contexts are created is also firing off a `factory.newInstance`, passing in just `script.getParams()` directly. I think this could use some overall cleanup as its somewhat convoluted.

Reply:
We discussed this issue in person and came to conclusion that this dead code is indeed dead code, and wouldn't make a difference if we started to use the parameters. Also we should fully remove the deprecated params.ctx support for 7.x (issue not logged yet).

Reply:
#36848 is the params.ctx removal, I saw this comment and decided to just nuke it.